andela/travel_tool_front
# #161374901 View travel readiness of potential travellers

#### What does this PR do?
-Enables Travel Admin to view travel readiness of visitors coming into their location

#### Description of Task to be completed?
-Write redux actions, reducers and middleware to fetch data from API and update the store
-Write Card Component to display travel readiness data from the store
-Update middleware to trigger file download when export button is clicked
#### How should this be manually tested?
- Clone this repo - git clone https://github.com/andela/travel_tool_front.git
- Switch to the repo directory - cd travel_tool_front
- Check out this branch - git checkout ft-view-potential-travellers-161374901
-Checkout the appropriate backend branch `ft-view-potential-travellers-161374901` and run `yarn:dbrollmigrate` and 
- Start the server - make start
- Install dependencies on the frontend - yarn install
- Start the server - yarn start
- Navigate to localhost:3000 in a browser
- Assign yourself the manager role and Travel Admin role so as to be able to approve your request in the database and view the dashboard page
- Navigate to the requests page and make a new request selecting your location as the destination.
- Navigate to the approvals page and approve your request
- Navigate to the dashboard page
- You should be able to view a card on the dashboard page with the title `Travel Readiness` that shows the visitors coming into your center, their % Completion in terms of travel readiness and their expected arrival date
- Click the export button on the card to download a csv format of the report

#### Any background context you want to provide?
This PR still has some failing tests. Kindly review the functionality as I work on fixing tests.

#### What are the relevant pivotal tracker stories?


#### Screenshots (if appropriate)
![ezgif com-crop](https://user-images.githubusercontent.com/37342753/47861291-873d2680-de03-11e8-8651-56d2e0686426.gif)



#### Questions:
None
